### ISSUE TITLE
False syntax errors with `associated_const_equality`

### ISSUE THREAD

**Issue by thorjelly** [rust-lang/rust-analyzer#11965]
The following code works in nightly with `cargo check` but rust-analyzer reports `HasCount<COUNT = {N}>` as containing syntax errors:

```rust
#![feature(associated_const_equality)]

trait HasCount {
    const COUNT: usize;
}

trait Foo {}

impl<T, const N: usize> Foo for T where T: HasCount<COUNT = {N}> {}
```

Using pre-release VSCode extension. The official Rust plugin works fine.

**Comment by memoryruins** [comment]
The rustc tracking issue is <https://github.com/rust-lang/rust/issues/92827>. It doesn't currently link to a doc/mcp/rfc describing the feature, but I imagine the syntax could be handled by r-a even before implementing lowering/analysis for them.

In the meantime, rust-analyzer syntax's built-in diagnostics can be [configured/disabled](https://rust-analyzer.github.io/manual.html#diagnostics) by users until the syntax is handled (if the current diagnostic is distracting for anyone).

`syn` similarly updated to handle this case recently <https://github.com/dtolnay/syn/commit/99560af0e7cac51c6482c56c89246bbbba3fdb33>.

**Comment by XFFXFF** [comment]
I'm working on this
